web/src/portico/signup.ts

```diff
@@ -27,6 +28,20 @@ $(() => {
             // the field yet.
             password_quality($(this).val()!, $("#pw_strength .bar"), $(this));
         });
+
+        $password_field.on("paste", function (e: JQuery.TriggeredEvent) {
+            // Paste in data even if it is longer than max length
+            // in order to allow the django default "Ensure this value
+            // has less than 100 characters" to be triggered, and avoid
+            // silent truncation.
+            assert(e.originalEvent instanceof ClipboardEvent);
+            const pastedData = e.originalEvent.clipboardData?.getData("text");
+
+            if (pastedData!.length > $password_field.data("maxLength")) {
+                e.preventDefault();
+                $(this).val(pastedData!);
+            }
+        });
```

Comment:
While this does the job and displays a warning (see below) when pasting a password longer than maxLength instead of silent truncation, I believe that this a pretty hacky solution, that would not make much sense without the comment. 

Would like to hear the reviewers’ thoughts on this, and any suggestions for a more naturally readable solution! Note: the code here is similar to [`web/src/input_pill`](https://github.com/zulip/zulip/blob/main/web/src/input_pill.ts).

Here's what the warning looks like on signup when pasting a 128 character password:
<img width="1552" alt="Screenshot 2024-03-26 at 3 02 26 PM" src="https://github.com/zulip/zulip/assets/72880238/ddc5cd85-1249-43f5-9fdc-b115ad91a551">